decoder/src/lib.rs

```diff
@@ -87,15 +87,24 @@ pub struct Table {
     entries: BTreeMap<usize, TableEntry>,
 }
 
+/// Checks if the version encoded in the symbol table is compatible with this version of the
+/// `decoder` crate
+pub fn check_version(version: &str) -> Result<(), String> {
+    if version != DEFMT_VERSION {
+        return Err(format!(
+            "defmt version mismatch (firmware is using {}, host is using {}); \
+             are you using the same git version of defmt and related tools?
+Try `cargo install`-ing the latest git version of `probe-run` AND updating your project's `Cargo.lock` with `cargo update`",
+            version, DEFMT_VERSION,
+        ));
+    }
+
+    Ok(())
+}
+
 impl Table {
     pub fn new(entries: BTreeMap<usize, TableEntry>, version: &str) -> Result<Self, String> {
-        if version != DEFMT_VERSION {
-            return Err(format!(
-                "defmt version mismatch (firmware is using {}, host is using {}); \
-                 are you using the same git version of defmt and related tools?",
-                version, DEFMT_VERSION,
-            ));
-        }
+        check_version(version)?;
```

Comment:
Seems like this could be removed now, and we could just always have the caller do it. This would allow skipping the version check too, which I've wanted a couple of times.

Reply:
will remove it and document that the caller should do `check_version`